ui/views_test.py

```diff
@@ -32,6 +35,27 @@ def test_program_liveness(self):
             status_code=200
         )
 
+    def test_unauthenticated_user_redirect(self):
+        """Verify that an unauthenticated user can't visit '/dashboard'"""
+        response = self.client.get(reverse('ui-dashboard'))
+        self.assertRedirects(
+            response,
+            "/?next=" + reverse('ui-dashboard')
```

Comment:
It would be better to use a format string here (ie: `/?next={}".format(url)`). Format strings are easier to work with since they will convert any argument to a string, and they're better for localization since the string being localized is in one piece, for example: http://pylonsbook.com/en/1.1/internationalization-and-localization.html#marking-strings-for-internationalization